src/query/service/src/pipelines/processors/transforms/hash_join/probe_state.rs

```diff
@@ -33,15 +34,14 @@ pub struct ProbeState {
 
 impl ProbeState {
     pub fn clear(&mut self) {
-        self.probe_indexes.clear();
         self.build_indexes.clear();
         self.row_state.clear();
         self.valids = None;
     }
 
     pub fn with_capacity(capacity: usize) -> Self {
         ProbeState {
-            probe_indexes: Vec::with_capacity(capacity),
+            probe_indexes: vec![(0, 0); JOIN_MAX_BLOCK_SIZE],
```

Comment:
Hmm, if just have two small table to join, do we need to allocate `JOIN_MAX_BLOCK_SIZE`?

Reply:
If just have two small table to join, we will still need to allocate `JOIN_MAX_BLOCK_SIZE` in the `probe_inner_join`.